UI/DamageIndicator.cs

```diff
@@ -0,0 +1,39 @@
+using Godot;
+
+
+public class DamageIndicator : TextureRect
+{
+	Vector2 ShotFirePosition;
+	float OriginalRotation;
+	public override void _Ready()
+	{
+		// Start transparency Tween in _Ready so we know the Tween has been created
+		Tween TransparencyTween = GetNode<Tween>("Tween");
+		// Tween the "modulate" property
+		TransparencyTween.InterpolateProperty(this, "modulate", new Color(255, 255, 255, 1), new Color(255, 255, 255, 0), 3, Tween.TransitionType.Linear, Tween.EaseType.InOut);
+		TransparencyTween.Start();
+	}
+
+
+	public void SetShotPosition(Vector3 ShotFireArg)
```

Comment:
Undescriptive argument name (and Arg is unneeded unless it has the same name as a variable or member)